clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java

```diff
@@ -200,8 +200,14 @@ public void promptPushPrimer(JSONObject jsonObject){
             //Checks whether permission request is asked for the first time.
             boolean isFirstTimeRequest = CTPreferenceCache.getInstance(context, config).isFirstTimeRequest();
 
+            Activity currentActivity = CoreMetaData.getCurrentActivity();
+            if (currentActivity == null) {
+                Logger.d("CurrentActivity reference is null. Can't process the push-primer request!");
```

Comment:
@shivamsharma2710 add more info in logs, info that we told app devs in the past, like make sure you have registered activity life cycle callbacks with CT, call push primer method in onresume etc.

Reply:
@piyush-kukadiya Added more details in logs.